app/src/main/java/de/grobox/transportr/utils/TransportrUtils.kt

```diff
@@ -114,3 +119,16 @@ object TransportrUtils {
 }
 
 fun Location.hasLocation() = hasCoord() && (latAs1E6 != 0 || lonAs1E6 != 0)
+
+fun updateGlobalHttpProxy(newProxy: Proxy, manager: TransportNetworkManager) {
+    // MapBox
+    HttpRequestUtil.setOkHttpClient(
+        OkHttpClient.Builder()
+            .proxy(newProxy)
+            .build())
+    // public-transport-enabler
+    manager.transportNetwork.value?.let {
+        if (it.networkProvider is AbstractNetworkProvider)
+            (it.networkProvider as AbstractNetworkProvider).setProxy(newProxy)
+    }
```

Comment:
Should we maybe add the proxy to PTE where its provider gets created instead?

Reply:
I've tried to do that at first but it seems like the providers are created in a file called `de.grobox.transportr.networks.TransportNetworks` as part of a private val called `networks`, so I'm not sure where to take the proxy configuration from in that case. (Unless I've got something mixed up?)

It would work though (I've done it with hard-coded values) and I agree, it would probably be cleaner.

Reply:
Maybe add the proxy settings as a lamda parameter here:

https://github.com/grote/Transportr/blob/7b1866e15ae3cb26a31bb69195bc80dcf51c30c1/app/src/main/java/de/grobox/transportr/networks/TransportNetwork.kt#L42

Reply:
I've tried to implement something like this now but I'm having some problems with it. To not spam this thread, I've made the commit in a new branch: https://github.com/T0astBread/Transportr/commit/4bb422b207aac951530a372bce3ecaf58fcb1f49.

That commit crashes when I do a location search with the following stacktrace:
```
2020-11-17 13:11:12.395 9655-9655/de.grobox.liberario.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: de.grobox.liberario.debug, PID: 9655
    kotlin.UninitializedPropertyAccessException: lateinit property settingsManager has not been initialized
        at de.grobox.transportr.locations.LocationView.getSettingsManager$app_debug(LocationView.kt:66)
        at de.grobox.transportr.locations.LocationView$startSuggestLocationsTaskDelayed$1.run(LocationView.kt:239)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

I think there are a few places in the code where dependency injection won't run before a SettingsManager is needed for the proxy and I'm not really sure what to do about that.

Reply:
Oh and I've also not made the proxy a parameter to that factory function directly because you'd have to copy+paste `setProxy` in the `TransportNetworks` file for every provider in that case.

Reply:
Injecting stuff in Views might be trickier than in Activities and Fragments. Thanks for trying. If the first solution also handles network changes, I'd be fine with me, but let's also see what @ialokim thinks.

Reply:
I've just tried, it seems to work across network changes

Reply:
I think that would be fine with me too, as a network change triggers the MainActivity to be recreated also the proxy settings are applied.